presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/csv/CsvRowEncoder.java

```diff
@@ -0,0 +1,151 @@
+    private String[] row;
+
+    public CsvRowEncoder(ConnectorSession session, List<EncoderColumnHandle> columnHandles)
+    {
+        super(session, columnHandles);
+        this.size = columnHandles.size();
+        this.row = new String[this.size];
+    }
+
+    @Override
+    protected List<EncoderColumnHandle> validateColumns(List<EncoderColumnHandle> columnHandles)
+    {
+        for (EncoderColumnHandle columnHandle : columnHandles) {
+            checkArgument(columnHandle.getFormatHint() == null, "unexpected format hint '%s' defined for column '%s'", columnHandle.getFormatHint(), columnHandle.getName());
+            checkArgument(columnHandle.getDataFormat() == null, "unexpected data format '%s' defined for column '%s'", columnHandle.getDataFormat(), columnHandle.getName());
+
+            checkArgument(isSupportedType(columnHandle.getType()), "unsupported column type '%s' for column '%s'", columnHandle.getType(), columnHandle.getName());
+        }
+        return ImmutableList.copyOf(columnHandles);
+    }
+
+    private boolean isSupportedType(Type type)
+    {
+        return isVarcharType(type) || SUPPORTED_PRIMITIVE_TYPES.contains(type);
+    }
+
+    @Override
+    protected void appendNullValue()
+    {
+        row[currentColumnIndex] = null;
+    }
+
+    @Override
+    protected void appendLong(long value)
+    {
+        row[currentColumnIndex] = Long.toString(value);
+    }
+
+    @Override
+    protected void appendInt(int value)
+    {
+        row[currentColumnIndex] = Integer.toString(value);
+    }
+
+    @Override
+    protected void appendShort(short value)
+    {
+        row[currentColumnIndex] = Short.toString(value);
+    }
+
+    @Override
+    protected void appendByte(byte value)
+    {
+        row[currentColumnIndex] = Byte.toString(value);
+    }
+
+    @Override
+    protected void appendDouble(double value)
+    {
+        row[currentColumnIndex] = Double.toString(value);
+    }
+
+    @Override
+    protected void appendFloat(float value)
+    {
+        row[currentColumnIndex] = Float.toString(value);
+    }
+
+    @Override
+    protected void appendBoolean(boolean value)
+    {
+        row[currentColumnIndex] = Boolean.toString(value);
+    }
+
+    @Override
+    protected void appendString(String value)
+    {
+        row[currentColumnIndex] = value;
+    }
+
+    @Override
+    public byte[] toByteArray()
+    {
+        try (ByteArrayOutputStream byteArrayOuts = new ByteArrayOutputStream();
+                OutputStreamWriter outsWriter = new OutputStreamWriter(byteArrayOuts, StandardCharsets.UTF_8);
+                CSVWriter writer = new CSVWriter(outsWriter, ',', '"', "")) {
+            writer.writeNext(row);
+            writer.flush();
+            return byteArrayOuts.toByteArray();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    protected void clear()
+    {
+        row = new String[size];
```

Comment:
not really needed right?

Reply:
If we make sure that every index is updated for every row then no it is not needed